tachyon/zk/base/prover.h

```diff
@@ -10,13 +10,14 @@
 #include <memory>
 #include <utility>
 
+#include "tachyon/math/polynomials/univariate/univariate_evaluation_domain.h"
 #include "tachyon/zk/base/blinded_polynomial.h"
 #include "tachyon/zk/base/blinder.h"
 #include "tachyon/zk/transcript/transcript.h"
 
 namespace tachyon::zk {
 
-template <typename _PCSTy>
+template <typename _PCSTy, size_t ExtendedMaxDegree>
```

Comment:
We agreed not to declare implementations like `UnivariateXXX` directly at the `Prover` level. Therefore, please consider changing from `size_t ExtendedMaxDegree` to `typename ExtendedDomain` in this part.
This is the content recently discussed with team lead.